File: app/src/main/java/me/imzack/app/end/injector/module/ReminderPresenterModule.java
package me.imzack.app.end.injector.module;

import me.imzack.app.end.view.contract.ReminderViewContract;

import dagger.Module;
import dagger.Provides;

@Module
public class ReminderPresenterModule {

    private final ReminderViewContract mReminderViewContract;
    private final int mPlanListPosition;

    public ReminderPresenterModule(ReminderViewContract reminderViewContract, int planListPosition) {
        mReminderViewContract = reminderViewContract;
        mPlanListPosition = planListPosition;
    }

    @Provides
    ReminderViewContract provideReminderViewContract() {
        return mReminderViewContract;
    }

    @Provides
    int providePlanListPosition() {
        return mPlanListPosition;
    }
}

File: app/src/main/kotlin/me/imzack/app/end/injector/module/ReminderPresenterModule.kt
package me.imzack.app.end.injector.module

import dagger.Module
import dagger.Provides
import me.imzack.app.end.view.contract.ReminderViewContract

@Module
class ReminderPresenterModule(private val mReminderViewContract: ReminderViewContract, private val mPlanListPosition: Int) {

    @Provides
    fun provideReminderViewContract() = mReminderViewContract

    @Provides
    fun providePlanListPosition() = mPlanListPosition
}
